sklearn/preprocessing/data.py

```diff
@@ -1744,6 +1744,37 @@ def __init__(self, n_values="auto", categorical_features="all",
         self.sparse = sparse
         self.handle_unknown = handle_unknown
 
+    def get_feature_names(self, input_features=None):
+        """
+        Return feature names for output features
+
+        Parameters
+        ----------
+        input_features : list of string, length n_features, optional
+            String names for input features if available. By default,
+            "x0", "x1", ... "xn_features" is used.
+
+        Returns
+        -------
+        output_feature_names : list of string, length sum(self.n_values_)
+
+        """
+        check_is_fitted(self, 'n_values_')
+
+        if input_features is None:
+            input_features = ['x%d' % i for i in range(len(self.n_values_))]
+        else:
+            if len(input_features) != len(self.n_values_):
+                raise ValueError("Number of input_features must equal to "
```

Comment:
This is clunky still. How about `Length of input_features is {0} but it must equal number of features when fitted: {1}.`